### PR TITLE
fix: address various issues with SSR

### PR DESCRIPTION
**Pull request by kyubisation**

### CODE REVIEW

File: src/components/navigation/navigation/navigation.e2e.ts [hunk 1]
```diff
@@ -275,6 +275,8 @@ describe('sbb-navigation', () => {
     await waitForCondition(() => didOpenEventSpy.events.length === 1);
     expect(didOpenEventSpy.count).to.be.equal(1);
     await waitForLitRender(element);
+    // TODO: It is not clear, why this is necessary here.
+    await aTimeout(10);
```
Comment (jeripeierSBB) [inline src/components/navigation/navigation/navigation.e2e.ts:279]: Still needed?

File: src/components/train/train-formation/train-formation.ts [hunk 2]
```diff
@@ -123,6 +124,10 @@ export class SbbTrainFormation extends LitElement {
     this._contentResizeObserver.disconnect();
     this._formationDiv = el;
     this._contentResizeObserver.observe(this._formationDiv as Element);
+    // There seems to be a slight difference between browser, in how the
+    // observer is called. In order to be consistent across browsers
+    // we set the width manually with a tick delay.
+    setTimeout(() => this._applyCssWidth());
```
Comment (jeripeierSBB) [inline src/components/train/train-formation/train-formation.ts:130]: I hope this works reliable ;-)

**Review by jeripeierSBB — APPROVED** [review]
LGTM, one question

File: src/components/train/train-formation/train-formation.ts (hunk 2 above)
Comment (kyubisation) [reply src/components/train/train-formation/train-formation.ts:130]: I hope so too, but the test is far more stable this way.

Comment (kyubisation) [reply src/components/train/train-formation/train-formation.ts]: Using updateComplete now

File: src/components/navigation/navigation/navigation.e2e.ts (hunk 1 above)
Comment (kyubisation) [reply src/components/navigation/navigation/navigation.e2e.ts]: No, thanks 👍

**Pull request merged**